frontend/app/src/people/widgetViews/organization/AssignUserRole.tsx

```diff
@@ -0,0 +1,267 @@
+`;
+
+const Username = styled.p`
+  color: #3c3f41;
+  text-align: center;
+  font-family: 'Barlow';
+  font-size: 1.25rem;
+  font-style: normal;
+  font-weight: 500;
+  line-height: 1.625rem;
+  margin-top: 0.69rem;
+  margin-bottom: 0;
+  text-transform: capitalize;
+`;
+
+const UserRolesContainer = styled.div`
+  padding: 3.25rem 3rem 3rem 3rem;
+  margin-top: 3.25rem;
+`;
+
+const UserRolesTitle = styled.h2`
+  color: #3c3f41;
+  font-family: 'Barlow';
+  font-size: 1.625rem;
+  font-style: normal;
+  font-weight: 800;
+  line-height: 1.625rem;
+  margin-bottom: 2.81rem;
+`;
+
+const RolesContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const RoleContainer = styled.div`
+  display: flex;
+  align-items: center;
+  margin-bottom: 1rem;
+`;
+
+const Checkbox = styled.input`
+  margin-right: 1rem;
+  width: 1rem;
+  height: 1rem;
+`;
+
+const Label = styled.label`
+  margin-bottom: 0;
+  color: #1e1f25;
+  font-family: 'Barlow';
+  font-size: 0.9375rem;
+  font-style: normal;
+  font-weight: 500;
+  line-height: 1.125rem;
+`;
+
+const AssingUserBtn = styled.button`
+  height: 3rem;
+  padding: 0.5rem 1rem;
+  width: 100%;
+  border-radius: 0.375rem;
+  font-family: 'Barlow';
+  font-size: 0.9375rem;
+  font-style: normal;
+  font-weight: 500;
+  line-height: 0rem;
+  letter-spacing: 0.00938rem;
+  background: #618aff;
+  box-shadow: 0px 2px 10px 0px rgba(97, 138, 255, 0.5);
+  border: none;
+  margin-top: 3rem;
+  color: #fff;
+  &:disabled {
+    border: 1px solid rgba(0, 0, 0, 0.07);
+    background: rgba(0, 0, 0, 0.04);
+    color: rgba(142, 150, 156, 0.85);
+    cursor: not-allowed;
+    box-shadow: none;
+  }
+`;
+
+const AssignUserRoles = (props: AssignUserModalProps) => {
+  const { isOpen, close, loading, user, setLoading, onSubmit, addToast } = props;
+  const { main } = useStores();
+
+  const config = nonWidgetConfigs['organizationusers'];
+
+  const isMobile = useIsMobile();
+
+  const roleData = main.bountyRoles.map((role: any) => ({
+    name: role.name,
+    status: false
+  }));
+
+  const [rolesData, setRolesData] = useState(roleData);
+  const [displayedRoles, setDisplayedRoles] = useState(s_RolesCategories);
+
+  useEffect(() => {
+    // Set default data roles for first assign user
+    const defaultRole = {
+      'Manage bounties': true,
+      'Fund organization': true,
+      'Withdraw from organization': true,
+      'View transaction history': true
+    };
+
+    const tempDataRole: { [id: string]: boolean } = {};
+
+    const newDisplayedRoles = displayedRoles.map((role: RolesCategory) => {
+      if (defaultRole[role.name]) {
+        role.status = true;
+        role.roles.forEach((dataRole: string) => (tempDataRole[dataRole] = true));
+      }
+      return role;
+    });
+
+    setRolesData((prev: BackendRoles[]) =>
+      prev.map((role: BackendRoles) => {
+        if (tempDataRole[role.name]) {
+          role.status = true;
+        }
+        return role;
+      })
```

Comment:
This should be testable aswell

Reply:
This might be a little difficult to test out because `tempDataRole` is an object created inside the function, and when using setState and getting the previous data i don't think we can pass the `tempDataRole` as a parameter.